From: "Brian J. Tarricone" <brian@tarricone.org>
Date: Sat, 21 Jan 2023 12:45:09 -0800
Subject: Take UI scale into acount when loading "image-missing" fallback

Fallback icons loaded were loading at half size (for a 2x scale) and for
cases where it mattered, were being positioned off-center.

Origin: upstream, 3.24.37, commit:89c7d5fe4b
---
 gtk/gtkiconhelper.c | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/gtk/gtkiconhelper.c b/gtk/gtkiconhelper.c
index f6e3e29..e51f8dc 100644
--- a/gtk/gtkiconhelper.c
+++ b/gtk/gtkiconhelper.c
@@ -484,11 +484,12 @@ ensure_surface_for_gicon (GtkIconHelper    *self,
   if (destination == NULL)
     {
       GError *error = NULL;
-      destination = gtk_icon_theme_load_icon (icon_theme,
-                                              "image-missing",
-                                              width,
-                                              flags | GTK_ICON_LOOKUP_USE_BUILTIN | GTK_ICON_LOOKUP_GENERIC_FALLBACK,
-                                              &error);
+      destination = gtk_icon_theme_load_icon_for_scale (icon_theme,
+                                                        "image-missing",
+                                                        MIN (width, height),
+                                                        scale,
+                                                        flags | GTK_ICON_LOOKUP_USE_BUILTIN | GTK_ICON_LOOKUP_GENERIC_FALLBACK,
+                                                        &error);
       /* We include this image as resource, so we always have it available or
        * the icontheme code is broken */
       g_assert_no_error (error);
